crates/hir-def/src/nameres.rs

```diff
@@ -114,6 +114,8 @@ pub struct DefMap {
     registered_attrs: Vec<SmolStr>,
     /// Custom tool modules registered with `#![register_tool]`.
     registered_tools: Vec<SmolStr>,
+    /// Unstable features of Rust enabled with `#![feature(A, B)]`.
+    unstable_features: FxHashSet<SmolStr>,
```

Comment:
I think we can just go with a `Vec` here, there aren't gonna be too many features so searching a Vec seems fine